lib/codecs/src/gelf.rs

```diff
@@ -65,7 +65,13 @@ pub(crate) static GELF_TARGET_PATHS: Lazy<GelfTargetPaths> = Lazy::new(|| GelfTa
     short_message: OwnedTargetPath::event(owned_value_path!(gelf_fields::SHORT_MESSAGE)),
 });
 
-/// Regex for matching valid field names. Must contain only word chars, periods and dashes.
-/// Additional field names must also be prefixed with an `_` , however that is intentionally
-/// omitted from this regex to be checked separately to create a specific error message.
+/// Regex for matching valid field names in the encoder. Must contain only word chars, periods
+/// and dashes. Additional field names must also be prefixed with an `_` , however that is
+/// intentionally omitted from this regex to be checked separately to create a specific
+/// error message.
 pub static VALID_FIELD_REGEX: Lazy<Regex> = Lazy::new(|| Regex::new(r"^[\w\.\-]*$").unwrap());
+
+/// Regex for matching valid field names in the decoder. Additionally, allow more characters
+/// (include `@`) possibly sent by the Graylog GELF Output, which not only produces field names matching
+/// VALID_FIELD_REGEX, but in fact any field names in Events present by Graylog.
+pub static VALID_DECODE_FIELD_REGEX: Lazy<Regex> = Lazy::new(|| Regex::new(r"^[\w\.\-@]*$").unwrap());
```

Comment:
Let's keep the same regex for both encoder and decoder.

Reply:
+1

Reply:
done, and pushed.